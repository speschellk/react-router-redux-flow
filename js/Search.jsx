// @flow

import React from 'react';
import { connect } from 'react-redux';
import ShowCard from './ShowCard';
import Header from './Header';

const Search = (props: { searchTerm: string, shows: Array<Show> }) => (
  <div className="search">
    <Header
      searchTerm={props.searchTerm}
      showSearch
    />
    <div>
      {props.shows
        .filter(
          show =>
            `${show.title} ${show.description}`
              .toUpperCase()
              .indexOf(props.searchTerm.toUpperCase()) >= 0
        )
        .map(show => <ShowCard key={show.imdbID} {...show} />)}
    </div>
  </div>
);

// return object whose properties will be spread over Search
// passes props into search
const mapStateToProps = state => ({
  searchTerm: state.searchTerm
});

// only for testing purposes
export const Unwrapped = Search;
export default connect(mapStateToProps)(Search);
