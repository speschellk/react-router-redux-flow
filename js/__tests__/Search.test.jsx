import React from 'react';
import Search from '../Search';
import { shallow } from 'enzyme'; // stubs out all the children components

test('Search renders correctly', () => {
  const component = shallow(<Search />);
  expect(component).toMatchSnapshot();
});
