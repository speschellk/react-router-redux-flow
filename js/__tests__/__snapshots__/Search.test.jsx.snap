// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <header>
    <h1>
      svideo
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </header>
  <div>
    <ShowCard
      description="Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; \\"Earnest 'Earn' Marks,\\" an ambitious college drop-out and his estranged cousin, who suddenly becomes a star."
      poster="a.jpg"
      title="Atlanta"
      year="2008–2013"
    />
    <ShowCard
      description="U.S. Attorney Chuck Rhoades goes after hedge fund king, Bobby \\"Axe\\" Axelrod in a battle between two powerful New York figures."
      poster="b.jpg"
      title="Billions"
      year="2016–"
    />
    <ShowCard
      description="A television anthology series that shows the dark side of life and technology."
      poster="bm.jpg"
      title="Black Mirror"
      year="2011–"
    />
    <ShowCard
      description="A high school chemistry teacher diagnosed with inoperable lung cancer turns to manufacturing and selling methamphetamine in order to secure his family's future."
      poster="bb.jpg"
      title="Breaking Bad"
      year="2008–2013"
    />
    <ShowCard
      description="Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years."
      poster="got.jpg"
      title="Game of Thrones"
      year="2011–"
    />
    <ShowCard
      description="A bipolar CIA operative becomes convinced a prisoner of war has been turned by al-Qaeda and is planning to carry out a terrorist attack on American soil."
      poster="h.jpg"
      title="Homeland"
      year="2011–"
    />
    <ShowCard
      description="A Congressman works with his equally conniving wife to exact revenge on the people who betrayed him."
      poster="hoc.jpg"
      title="House of Cards"
      year="2013–"
    />
    <ShowCard
      description="The personal and professional life of Dev, a 30-year-old actor in New York."
      poster="mon.jpg"
      title="Master of None"
      year="2015–"
    />
    <ShowCard
      description="A chronicled look at the criminal exploits of Colombian drug lord Pablo Escobar."
      poster="n.jpg"
      title="Narcos"
      year="2015–"
    />
    <ShowCard
      description="The story of Piper Chapman, a woman in her thirties who is sentenced to fifteen months in prison after being convicted of a decade-old crime of transporting money for her drug-dealing girlfriend."
      poster="oitnb.jpg"
      title="Orange Is the New Black"
      year="2013–"
    />
    <ShowCard
      description="Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper."
      poster="sv.jpg"
      title="Silicon Valley"
      year="2014–"
    />
    <ShowCard
      description="When a young boy disappears, his mother, a police chief, and his friends must confront terrifying forces in order to get him back."
      poster="st.jpg"
      title="Stranger Things"
      year="2016–"
    />
    <ShowCard
      description="Two Soviet intelligence agents pose as a married couple to spy on the American government."
      poster="ta.jpg"
      title="The Americans"
      year="2013–"
    />
    <ShowCard
      description="Former Senator Selina Meyer finds that being Vice President of the United States is nothing like she hoped and everything that everyone ever warned her about."
      poster="v.jpg"
      title="Veep"
      year="2012–"
    />
    <ShowCard
      description="Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence."
      poster="ww.jpg"
      title="Westworld"
      year="2016–"
    />
  </div>
</div>
`;
